### PR TITLE
Rename `setSparse()` -> `sparse()` and make it async

### PR DESCRIPTION
**Pull request by kasperisager**
Let me know if you think `setSparse()` is a better fit also for the async version.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -92,10 +92,24 @@ exports.punchHole = function punchHole (fd, offset, length) {
...
+  }
+
+  const promise = new Promise((resolve, reject) => {
+    ctx.resolve = resolve
+    ctx.reject = reject
+  })
+
+  const errno = binding.fsctl_napi_sparse(req, fd, ctx, onwork)
 
   if (errno < 0) throw toError(errno)
```
Comment (mafintosh) [inline index.js]: return Promise.reject(toError(errno))

**Comment by mafintosh** [comment]
I like sparse!

File: index.js (hunk 1 above)
Comment (kasperisager) [reply index.js]: Let me fix the other one as well 👍

File: index.js [hunk 2]
```diff
@@ -87,15 +87,29 @@ exports.punchHole = function punchHole (fd, offset, length) {
...
 
-  if (errno < 0) throw toError(errno)
+  if (errno < 0) return Promise.reject(toError(errno))
 
   return promise
 }
 
-exports.setSparse = function setSparse (fd) {
-  const errno = binding.fsctl_napi_set_sparse(fd)
+exports.sparse = function sparse (fd) {
```
Comment (mafintosh) [inline index.js:95]: should we add `if (process.platform !== 'win32') return Promise.resolve()` ?

**Pull request merged**